tests/phpunit/tests/canonical.php

```diff
@@ -425,4 +425,51 @@ public function test_canonical_attachment_page_redirect_with_option_disabled() {
 
 		$this->assertSame( $expected, $url );
 	}
+
+	/**
+	 * @ticket 4463
+	 * @dataProvider data_test_redirect_pretty_search_results
+	 */
+	public function test_redirect_pretty_search_results( $input_url, $expected ) {
+		$this->set_permalink_structure( '/%postname%/' );
+
+		$this->go_to( $input_url );
+		$this->assertSame(
+			sprintf( 'http://%1$s%2$s', WP_TESTS_DOMAIN, $expected ),
+			redirect_canonical( sprintf( 'http://%1$s%2$s', WP_TESTS_DOMAIN, $input_url ), false )
+		);
+	}
+
+	/**
+	 * Data provider for test_redirect_guess_404_permalink_with_custom_statuses().
+	 *
+	 * return array[] {
+	 *    array Arguments used to register custom status
+	 *    bool  Whether the 404 link is expected to redirect
+	 * }
+	 */
+	public function data_test_redirect_pretty_search_results() {
```

Comment:
I love the switch to the data provider, but this is a lot less test cases then exist currently.  I think including each of the feed types is helpful for ensuring each of them works as expected. I also think we need to keep the `test_search_encoded_chars` as that was added when a bug was fixed. 

Additionally, I think that if run on it's own it will fail since the setup that was in `test_search_paged` isn't done.